larpandoracontent/LArMonitoring/EventClusterValidationAlgorithm.cc

```diff
@@ -151,181 +249,681 @@ void EventClusterValidationAlgorithm::GetHitParents(
         }
     }
 
+    if (m_handleDeltaRays)
+    {
+        for (auto &[pCaloHit, parents] : hitParents)
+        {
+            parents.m_pMainMC = this->FoldPotentialDeltaRayTo(pCaloHit, parents.m_pMainMC);
+        }
+    }
+
     for (const Cluster *const pCluster : clusters)
     {
         const CaloHitList &isolatedHits{pCluster->GetIsolatedCaloHitList()};
         CaloHitList clusterCaloHits;
         pCluster->GetOrderedCaloHitList().FillCaloHitList(clusterCaloHits);
         clusterCaloHits.insert(clusterCaloHits.end(), isolatedHits.begin(), isolatedHits.end());
+
         for (const CaloHit *const pCaloHit : clusterCaloHits)
         {
-            // Ignoring the calo hit if truth matching is missing
             if (hitParents.find(pCaloHit) == hitParents.end())
+            {
```

Comment:
The now deleted comment notes that hits with MC are ignored (which makes sense), but, if I understand correctly, hitParents can now have hits removed due to other criteria, is it still correct to skip over these hits, or do they need to be accounted for in the metrics (e.g. if I've understood correctly, a hit with a true PDG code that doesn't fit the category you are looking at would be omitted from the metrics, when in practice I think you'd want this to turn up in the error accounting. Have I understood this correctly?)

Reply:
Yes there are more configurable reasons to drop hits from hitParents map now (hit is in a null reco cluster, minimum required number of MC hits in a view). These hits are only skipped over if configured to do so which I think is ok.  The only ones that are completely ignored is when the hit has no MC weight.